File: projects/examples/python/error_catching.py
# Built-In Modules
import random

# This Is A Try, Except Block
# This Will Try To Run The Code In The Try Block
try:
    # This Will Throw An Error Because You Can't Divide By Zero
    print(1/0)
# This Will Catch The Error And Run Code Placed Here
# Using Except Without A Specific Error Will Catch Any Error
except:
    print("There Was An Error")

# You Can Also Catch Specific Errors
try:
    # This Will Throw An Error
    print(1/0)
# This Will Catch The Error
except ZeroDivisionError:
    print("There Was A Zero Division Error")

# You Can Also Catch Multiple Errors
try:
    print(1/0)
except (ZeroDivisionError, TypeError):
    print("There Was An Error")

# If You Want To Catch Different Errors You Can Use Multiple Except Blocks
try:
    # This Will Throw An Error
    if random.random() > 0.5: # Will Pick At Random, Either 1/0 Or "Hello World!" + 1. Each Will Throw A Different Error
        print(1/0)
    else:
        print("Hello World!" + 1)
except ZeroDivisionError:
    print("There Was A Zero Division Error")
except TypeError:
    print("There Was A Type Error")

# You Can Also Use Else And Finally
try:
    if random.random() > 0.5:
        print(1/0)
    else:
        print("Hello World!")
except:
    print("There Was An Error")
else:
    print("There Was No Error")

File: archive/projects/examples/java/error_catching.java
package projects.examples.java;

public class error_catching {
    // This Is The Class

    public static void main(String[] args) {
        // This Is The Main Method

        // This Is A Try-Catch Block
        // This Is The Try Block, It Will Try To Run This Code
        try {
            int[] myNumbers = {1, 2, 3};
            System.out.println(myNumbers[10]);
            // This Will Throw An Error, Because There Is No Value At Index 10
        
        }
        // This Is The Catch Block, It Will Run If The Try Block Throws An Error
        catch (Exception e) {
            System.out.println("Something Went Wrong!");
        }


        // You Can Catch More Specific Errors As Well
        try {
            int[] myNumbers = {1, 2, 3};
            System.out.println(myNumbers[10]);
        }
        // This Is The Catch Block For ArrayIndexOutOfBoundsException, Which Is The Error That Will Be Thrown
        catch (ArrayIndexOutOfBoundsException e) {
            System.out.println("Something Went Wrong!");
        }

        // You Can Also Catch Multiple Errors In One Catch Block
        try {
            int[] myNumbers = {1, 2, 3};
            System.out.println(myNumbers[10]);
        }
        // This Is The Catch Block For ArrayIndexOutOfBoundsException And NullPointerException, Which Are The Errors That Will Be Thrown
        catch (ArrayIndexOutOfBoundsException | NullPointerException e) {
            System.out.println("Something Went Wrong!");
        }

        // You Can Also Use Them In Separate Catch Blocks
        try {
            int[] myNumbers = {1, 2, 3};
            System.out.println(myNumbers[10]);
        }
        // This Is The Catch Block For ArrayIndexOutOfBoundsException, Which Is The Error That Will Be Thrown
        catch (ArrayIndexOutOfBoundsException e) {
            System.out.println("Something Went Wrong!");
        }
        // This Is The Catch Block For NullPointerException, Which Is Another Error That Will Be Thrown
        catch (NullPointerException e) {
            System.out.println("Something Went Wrong!");
        }

        // You Can Print The Error Message With The Exception Object
        try {
            int[] myNumbers = {1, 2, 3};
            System.out.println(myNumbers[10]);
        }
        // This Is The Catch Block, It Will Run If The Try Block Throws An Error
        catch (Exception e) {
            System.out.println("Something Went Wrong!");
            System.out.println(e.getMessage());
        }

        // You Can Also Use A Finally Block
        // This Is The Try Block, It Will Try To Run This Code
        try {
            int[] myNumbers = {1, 2, 3};
            System.out.println(myNumbers[10]);
        }
        // This Is The Catch Block, It Will Run If The Try Block Throws An Error
        catch (Exception e) {
            System.out.println("Something Went Wrong!");
        }
        // This Is The Finally Block, It Will Run No Matter What
        finally {
            System.out.println("The Final Try-Catch Block Finished!");
        }
    }
}
